### PR TITLE
Datatable add

### PR DESCRIPTION
**Pull request by MohimMolla**
prefix crud  & show data into table add   Data table, script, necessary cdn, table, form

### CODE REVIEW

File: app/Http/Controllers/Backend/PrefixController.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        return view('backend.prefix.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        try {
+            $authId = auth()->id();
```
Comment (mrmoon007) [inline app/Http/Controllers/Backend/PrefixController.php:39]: Please add the request validation class. For more info please read this https://laravel.com/docs/11.x/validation#creating-form-requests

File: app/Http/Controllers/Backend/PrefixController.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    public function show(string $id)
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     */
+    public function edit(string $id)
+    {
```
Comment (mrmoon007) [inline app/Http/Controllers/Backend/PrefixController.php:78]: Please error handle. `$prefix = Prefix::findOrFail($id);
if( !$refix ) {
      return back()->withErrors(['error' => 'Request data not found. ']);
}
        return view('backend.prefix.edit', compact('prefix'));`

File: app/Http/Controllers/Backend/PrefixController.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, string $id)
+    {
+
+        try {
+            $prefix = Prefix::findOrFail($id);
+
+
```
Comment (mrmoon007) [inline app/Http/Controllers/Backend/PrefixController.php:92]: Same as https://github.com/mrmoon007/family-accommodation_11/pull/1/files#r1855367466

File: database/migrations/2024_11_11_041351_create_prefixes_table.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('prefixes', function (Blueprint $table) {
+            $table->id();
+            $table->string('name', 100);
+            $table->string('slug', 255)->unique()->nullable();
```
Comment (mrmoon007) [inline database/migrations/2024_11_11_041351_create_prefixes_table.php:17]: Can you explain it why you use unique and nullable?

File: resources/views/backend/layouts/partials/css.blade.php [hunk 5]
```diff
@@ -22,3 +22,11 @@
     integrity="sha256-4MX+61mt9NVvvuPjUWdUdyfZfxSB1/Rf9WtqRHgG5S0=" crossorigin="anonymous"><!-- jsvectormap -->
 <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/jsvectormap@1.5.3/dist/css/jsvectormap.min.css"
     integrity="sha256-+uGLJmmTKOqBr+2E6KDYs/NRsHxSkONXFHUL0fy2O/4=" crossorigin="anonymous">
+
+{{-- font-awesome cdn --}}
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.6.0/css/all.min.css"
+    integrity="sha512-Kc323vGBEqzTmouAECnVceyQqyqdsSiqLQISBL29aUW4U/M7pSPA/gEUZQqv1cwx4OnYxTxve5UMg5GT6L4JJg=="
+    crossorigin="anonymous" referrerpolicy="no-referrer" />
+
```
Comment (mrmoon007) [inline resources/views/backend/layouts/partials/css.blade.php:30]: Please use local css.

File: resources/views/backend/layouts/partials/css.blade.php [hunk 6]
```diff
@@ -22,3 +22,11 @@
     integrity="sha256-4MX+61mt9NVvvuPjUWdUdyfZfxSB1/Rf9WtqRHgG5S0=" crossorigin="anonymous"><!-- jsvectormap -->
 <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/jsvectormap@1.5.3/dist/css/jsvectormap.min.css"
     integrity="sha256-+uGLJmmTKOqBr+2E6KDYs/NRsHxSkONXFHUL0fy2O/4=" crossorigin="anonymous">
```
Comment (mrmoon007) [inline resources/views/backend/layouts/partials/css.blade.php:24]: Please use local css. Do not use remote CSS or JS.

File: resources/views/backend/layouts/partials/script.blade.php [hunk 7]
```diff
@@ -6,6 +6,16 @@
 <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.min.js"
```
Comment (mrmoon007) [inline resources/views/backend/layouts/partials/script.blade.php:6]: Same as https://github.com/mrmoon007/family-accommodation_11/pull/1#discussion_r1855368153

File: resources/views/backend/prefix/create.blade.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    <!--begin::Container-->
+    <div class="container-fluid">
+        <!--begin::Row-->
+        <div class="row">
+            <div class="col-sm-6">
+                <h3 class="mb-0">Profile</h3>
+            </div>
+            <div class="col-sm-6">
+                <ol class="breadcrumb float-sm-end">
+                    <li class="breadcrumb-item"><a href="#">Home</a></li>
```
Comment (mrmoon007) [inline resources/views/backend/prefix/create.blade.php:19]: Please link it Home page.

File: resources/views/backend/prefix/create.blade.php [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                            <label for="validationCustom05" class="form-label">Position</label>
+                            <input type="number" class="form-control" id="position" name="position" required>
+                            <div class="invalid-feedback">
+
+                            </div>
+                        </div>
+                        
+                    </div>
+                </div>
+                <div class="card-footer"> <button class="btn btn-info" type="submit">Submit form</button> </div>
```
Comment (mrmoon007) [inline resources/views/backend/prefix/create.blade.php:80]: Change button text like as Save.

File: resources/views/backend/prefix/edit.blade.php [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+                                <label for="name" class="form-label">Prefix Name</label>
+                                <input type="text" class="form-control" value="{{ $prefix->name ?? '' }}" id="name"
+                                    name="name">
+                                <div class="valid-feedback">Looks good!</div>
+                            </div>
+                            <div class="col-md-6">
+                                <label for="slug" class="form-label">Prefix Slug</label>
+                                <input type="text" class="form-control" id="slug" value="{{ $prefix->slug ?? '' }}"
+                                    name="slug">
+                                <div class="valid-feedback">Looks good!</div>
```
Comment (mrmoon007) [inline resources/views/backend/prefix/edit.blade.php:63]: Why use "looks goot!"?

File: resources/views/backend/prefix/edit.blade.php [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+                                <select class="form-select" id="status" name="status">
+                                    <option selected disabled value="">Choose statuse</option>
+                                    <option value="active"
+                                        {{ old('status', $prefix->status) == 'active' ? 'selected' : '' }}>Active</option>
+                                    <option value="inactive"
+                                        {{ old('status', $prefix->status) == 'inactive' ? 'selected' : '' }}>Inactive
+                                    </option>
+                                </select>
+                                <div class="invalid-feedback">
+                                    Please select a valid state.
```
Comment (mrmoon007) [inline resources/views/backend/prefix/edit.blade.php:80]: Why use "Please select a valid state"?

File: resources/views/backend/prefix/edit.blade.php [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+                                <input type="number" class="form-control" value="{{ $prefix->position ?? '' }}"
+                                    id="position" name="position">
+                                <div class="invalid-feedback">
+
+                                </div>
+                            </div>
+
+                        </div>
+                    </div>
+                    <div class="card-footer"> <button class="btn btn-info" type="submit">Submit form</button> </div>
```
Comment (mrmoon007) [inline resources/views/backend/prefix/edit.blade.php:94]: Please change text like as "Save".

File: app/Http/Controllers/Backend/PrefixController.php (hunk 1 above)
Comment (MohimMolla) [reply app/Http/Controllers/Backend/PrefixController.php:39]: done in feature/createAccount branch

File: app/Http/Controllers/Backend/PrefixController.php (hunk 2 above)
Comment (MohimMolla) [reply app/Http/Controllers/Backend/PrefixController.php:78]: done in feature/createAccount branch

File: app/Http/Controllers/Backend/PrefixController.php (hunk 3 above)
Comment (MohimMolla) [reply app/Http/Controllers/Backend/PrefixController.php:92]: done in feature/createAccount branch

File: resources/views/backend/prefix/create.blade.php (hunk 8 above)
Comment (MohimMolla) [reply resources/views/backend/prefix/create.blade.php:19]: its done

File: resources/views/backend/prefix/create.blade.php (hunk 9 above)
Comment (MohimMolla) [reply resources/views/backend/prefix/create.blade.php:80]: done

File: resources/views/backend/prefix/edit.blade.php (hunk 10 above)
Comment (MohimMolla) [reply resources/views/backend/prefix/edit.blade.php:63]: this is resolve use now i use  this 
 @error('name')
        <div class="invalid-feedback">{{ $message }}</div>
  @enderror

Comment (MohimMolla) [reply resources/views/backend/prefix/edit.blade.php:63]: @error('name')
<div class="invalid-feedback">{{ $message }}</div>
@enderror

File: resources/views/backend/prefix/edit.blade.php (hunk 11 above)
Comment (MohimMolla) [reply resources/views/backend/prefix/edit.blade.php:80]: now i use this

   @error('status')
                                            <div class="invalid-feedback">{{ $message }}</div>
                                        @enderror

File: resources/views/backend/layouts/partials/css.blade.php (hunk 5 above)
Comment (MohimMolla) [reply resources/views/backend/layouts/partials/css.blade.php:30]: done

File: resources/views/backend/layouts/partials/css.blade.php (hunk 6 above)
Comment (MohimMolla) [reply resources/views/backend/layouts/partials/css.blade.php:24]: done

File: resources/views/backend/layouts/partials/script.blade.php (hunk 7 above)
Comment (MohimMolla) [reply resources/views/backend/layouts/partials/script.blade.php:6]: done

**Pull request closed without merging**